pkg/utils/labels/labels.go

```diff
@@ -53,3 +53,16 @@ func DefaultIfEmpty(m map[string]string, def map[string]string) map[string]strin
 	}
 	return m
 }
+
+func LabelNamesEqual(a, b[]string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if a[i] != b[i] {
```

Comment:
Are these array sorted?

Reply:
well, in some cases they are sorted(at the beginning). but after updating a collector with new labels, they aren't sorted anymore. Maybe it makes sense to make a collector's update a bit more elaborated and also sort its labelNames slice after adding new labels. It would require some extra work to relabel all the metrics in the collector, though...